app/lib/actions/katello/content_view/promote_to_environment.rb

```diff
@@ -46,10 +46,14 @@ def rescue_strategy_for_self
 
         def finalize
           # update errata applicability counts for all hosts in the CV & LE
-          ::Katello::Host::ContentFacet.where(:content_view_id => input[:content_view_id],
-                                              :lifecycle_environment_id => input[:environment_id]).each do |facet|
-            facet.update_applicability_counts
-            facet.update_errata_status
+          Location.no_taxonomy_scope do
+            User.as_anonymous_admin do
```

Comment:
It seems like this fix works even without this `User.as_anonymous_admin` block. What does this block do?

Reply:
Great question!  To fix this bug, its not needed.  However a similar bug would happen if some user has permissions to publish/promote a content view but does not have permissions to view some set of hosts.  

The location/organization and current user are preserved in the task, so the task would get execute that code as that uesr and hit the same issue as being in the wrong location